### PR TITLE
golint: lower min_confidence

### PR DESCRIPTION
**Pull request by greut**

### CODE REVIEW

File: nics.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 // ListRequest build a ListNics request from the given Nic
 func (nic *Nic) ListRequest() (ListCommand, error) {
 	if nic.VirtualMachineID == "" {
-		return nil, fmt.Errorf("ListNics command requires the VirtualMachineID field to be set")
+		return nil, fmt.Error("command ListNics requires the VirtualMachineID field to be set")
```
Comment (pierre-emmanuelJ) [inline nics.go]: dont't remove the ```f```from ```fmt.Errorf```

File: request.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func (e *ErrorResponse) Error() string {
 // Success computes the values based on the RawMessage, either string or bool
 func (e *booleanResponse) IsSuccess() (bool, error) {
 	if e.Success == nil {
-		return false, fmt.Errorf("Not a valid booleanResponse")
+		return false, fmt.Error("not a valid booleanResponse, Success is missing")
```
Comment (pierre-emmanuelJ) [inline request.go]: idem

File: security_groups.go [hunk 3]
```diff
@@ -17,11 +17,12 @@ func (*SecurityGroup) ResourceType() string {
...
 func (sg *SecurityGroup) Get(ctx context.Context, client *Client) error {
-	if sg.ID == "" && sg.Name == "" {
-		return fmt.Errorf("A SecurityGroup may only be searched using ID or Name")
+	req, err := sg.ListRequest()
+	if err != nil {
+		return err
 	}
 
-	resp, err := client.List(&SecurityGroup{Name: sg.Name, ID: sg.ID})
+	resp, err := client.List(req)
```
Comment (pierre-emmanuelJ) [inline security_groups.go]: i think ```[]interface{}``` is not listable

File: request.go (hunk 2 above)
Comment (greut) [reply request.go]: I should use `error.Error`, indeed..

**Comment by pierre-emmanuelJ** [comment]
👍

**Comment by pierre-emmanuelJ** [comment]
```
[src] Running shell script

+ golint -set_exit_status -min_confidence 0.6 github.com/exoscale/egoscale/cmd/cs

/go/src/github.com/exoscale/egoscale/cmd/cs/flags.go:90:21: error strings should not be capitalized or end with punctuation or a newline

/go/src/github.com/exoscale/egoscale/cmd/cs/flags.go:172:21: error strings should not be capitalized or end with punctuation or a newline

/go/src/github.com/exoscale/egoscale/cmd/cs/flags.go:205:21: error strings should not be capitalized or end with punctuation or a newline

Found 3 lint suggestions; failing.

script returned exit code 1
```

**Comment by greut** [comment]
oh snap!

File: Jenkinsfile [hunk 4]
```diff
@@ -55,12 +55,8 @@ def golint() {
...
     def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.10')
     image.inside("-u root --net=host -v ${env.WORKSPACE}/src:/go/src/github.com/exoscale/egoscale") {
-      sh 'golint -set_exit_status -min_confidence 0.6 github.com/exoscale/egoscale'
-      sh 'golint -set_exit_status -min_confidence 0.6 github.com/exoscale/egoscale/cmd/cs'
-      sh 'golint -set_exit_status -min_confidence 0.6 github.com/exoscale/egoscale/generate'
-      sh 'go vet github.com/exoscale/egoscale'
-      sh 'go vet github.com/exoscale/egoscale/cmd/cs'
-      sh 'go vet github.com/exoscale/egoscale/generate'
+      sh 'golint -set_exit_status -min_confidence 0.6 $(go list github.com/exoscale/egoscale/... | grep -v /vendor/)'
+      sh 'go vet $(github.com/exoscale/egoscale/... | grep -v /vendor/)'
```
Comment (greut) [inline Jenkinsfile]: @pierre-emmanuelJ this should catch everything.

**Comment by pierre-emmanuelJ** [comment]
👍

**Pull request merged**